src/Orchestration/Adapter.php

```diff
@@ -38,32 +36,68 @@ abstract public function pull(string $image): bool;
     /**
      * List Containers
      *
-     * @return bool
+     * @return array
      */
-    abstract public function list(): bool;
+    abstract public function list(): array;
+
+    /**
+     * Filter ENV vars
+     * 
+     * @param string $string
+     * 
+     * @return string
+     */
+    public function filterEnvKey(string $string): string
+    {
+        $string     = \str_split($string);
+        $output     = '';
+
+        foreach ($string as $char) {
+            if(\in_array($char, ['0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z','a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z','_',])) {
+                $output .= $char;
+            }
+        }
+    
+        return $output;
+    }
 
     /**
      * Run Container
+     * 
+     * @param string $image
+     * @param string $name
+     * @param string $entrypoint
+     * @param array $command
```

Comment:
Can we be more specific with arrays in docblocks? The more accurate these are (e.g. `string[]` instead of `array`), the better our static analysis tools (psalm) can catch possible errors.